### ISSUE TITLE
No error on Telegram agent - nothing in logs

### ISSUE THREAD

**Issue by GiantCrocodile** [huginn/huginn#2135]
I have this construct:

```
Website agent ->  Event Formatting Agent ->  Telegram Agent
```

I see that I get content from website and it gets formatted correctly at my second agent. My telegram agent receives the events from second agent but it seem to do nothing. I can't dry run my telegram agent nor are there any entries in log. How can I debug this or see what is wrong?

**Comment by dsander** [comment]
Hi,

you are right the Agent currently does not output any debug or error information. Did you verify your `auth_token` and `chat_id`?

`https://api.telegram.org/bot<auth_token>/getUpdates` should give you a list with the most recent messages the bot user received.

**Comment by GiantCrocodile** [comment]
Hey @dsander.

`auth_token` and `chat_id` are both correct. I verified this. I don't see any messages at `https://api.telegram.org/bot<auth_token>/getUpdates` except the ones I send manually to my bot from my Telegram client on Android.

**Comment by GiantCrocodile** [comment]
I found out the problem. My converter works with this input:

```
{
  "instructions": {
    "text": "test"
  },
  "mode": "clean"
}
```

but with nested keys under key `text` it doesn't work with Telegram bot together. It shows correct output in dry run and I expected Telegram bot to put them together as a plain string but it seems that Telegram bot just gives no info at all (not even when messages are sent out) in logs and can't process nested `text` keys.

**Comment by dsander** [comment]
That is correct, the Agent expects the received Events to have one of the supported keys, without them it did not do anything. I submitted a PR which improves the Agent a bit, it now logs on failure and is configurable without manually checking the Telegram API.

**Comment by GiantCrocodile** [comment]
What about `parse_mode` set to `html` logging no error? I have no idea why it doesn't send my message as html. It is just plain text with a `<br>` tag and an `img` tag. Right now this feels very unstable to me and leaves me behind with a lot of question marks in my brain.

**Comment by dsander** [comment]
Did you try the PR? It will log an error message: `{"ok"=>false, "error_code"=>400, "description"=>"Bad Request: can't parse entities in message text: Unsupported start tag \"br\" at byte offset 9"}`

If you are using docker you can use `huginnbuilder/huginn:pr-2138` or ` huginnbuilder/huginn-single-process:pr-2138` as your image.

**Comment by GiantCrocodile** [comment]
I will try this right now @dsander.

**Comment by GiantCrocodile** [comment]
I don't know why but when I use this Huginn version (`huginnbuilder/huginn:pr-2138`) my website agent does generate events but my converter agent do not get them.

**Comment by GiantCrocodile** [comment]
![grafik](https://user-images.githubusercontent.com/6271057/31047338-dbec2462-a608-11e7-97f2-ed5688cf85d1.png)

**Comment by GiantCrocodile** [comment]
I tried this input:
```json
  {
    "text": "some text <br> <img src='http://www.domain.tld/someimage.jpg' />"
  }
```

and I got this error:
```
[00:00:00] ERROR -- : {"ok"=>false, "error_code"=>400, "description"=>"Bad Request: can't parse entities in message text: Unsupported start tag \"br\" at byte offset 28"}
```

So why is this invalid input?

**Comment by dsander** [comment]
The pull request does not change the behavior of any other Agent. Are you sure the WebsiteAgent emitted new events?

**Comment by GiantCrocodile** [comment]
I'm sure @dsander. I'm always deleting all events before I run the agents again. The events in the screenshots are newly created ones. I'm not familiar with the way I did run Docker so I can't say if maybe someone's else commit change something. I just know that I stopped all three agents, deleted the events of them and set scheduling of website agent to never three days ago. After running them now, they are broken. I also changed, as you can see in screenshot above, the schedule for website agent back to daily.

**Comment by dsander** [comment]
Is there anything in the output of the docker container that might explain the problem?
